src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -261,18 +259,47 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
             return true;
         }
 
+        private void LoadThreadsAndAttach()
+        {
+            foreach (var tid in GetThreads(ProcessId))
+            {
+                var status = (int)ptrace(PTRACE_ATTACH, (int) tid, IntPtr.Zero, IntPtr.Zero);
+                if (status >= 0)
+                {
+                    status = waitpid((int)tid, IntPtr.Zero, 0);
+                }
+                if (status >= 0)
+                {
+                    // Remember the threads we managed to attach to
+                    _threadIDs.Add(tid);
+                }
+
+                // Only throw if we can't attach to the main thread as some threads could exit in the meantime
+                if (status < 0 && tid == ProcessId)
+                {
+                    var errno = Marshal.GetLastWin32Error();
+                    throw new ClrDiagnosticsException($"Could not attach to thread {tid}, errno: {errno}", errno);
```

Comment:
Is the `GetThreads(ProcessId)` guaranteed to return the main thread first in the enumerable? Can it happen that we attach to a thread and then don't detach because of the exception throw?

Reply:
All the threads we attached to are in the `_threadIDs`, and there's a finalizer so in the worst case it'll take care of them eventually.